File: flutter_hospital/lib/models/question.dart
// ============================================================
// models/question.dart
// نموذج بيانات الـ Question المحفوظة في الـ Database
// ============================================================

import 'dart:convert';

class Answer {
  final String question;
  final int answer;

  Answer({required this.question, required this.answer});

  factory Answer.fromJson(Map<String, dynamic> json) {
    return Answer(
      question: json['question'] as String,
      answer: json['answer'] as int,
    );
  }

  Map<String, dynamic> toJson() => {
        'question': question,
        'answer': answer,
      };

  Answer copyWith({String? question, int? answer}) {
    return Answer(
      question: question ?? this.question,
      answer: answer ?? this.answer,
    );
  }
}

class Item {
  final String title;
  List<Answer> subItems;

  Item({required this.title, required this.subItems});

  factory Item.fromJson(Map<String, dynamic> json) {
    return Item(
      title: json['title'] as String,
      subItems: (json['subItems'] as List)
          .map((e) => Answer.fromJson(e as Map<String, dynamic>))
          .toList(),
    );
  }

  Map<String, dynamic> toJson() => {
        'title': title,
        'subItems': subItems.map((e) => e.toJson()).toList(),
      };
}

class Question {
  final int? id;
  String checkList;
  List<Item> items;
  final int dateAdded;

  Question({
    this.id,
    required this.checkList,
    required this.items,
    required this.dateAdded,
  });

  factory Question.fromMap(Map<String, dynamic> map) {
    return Question(
      id: map['id'] as int?,
      checkList: map['checkList'] as String,
      items: (jsonDecode(map['items'] as String) as List)
          .map((e) => Item.fromJson(e as Map<String, dynamic>))
          .toList(),
      dateAdded: map['dateAdded'] as int,
    );
  }

  Map<String, dynamic> toMap() => {
        if (id != null) 'id': id,
        'checkList': checkList,
        'items': jsonEncode(items.map((e) => e.toJson()).toList()),
        'dateAdded': dateAdded,
      };

  Question copyWith({
    int? id,
    String? checkList,
    List<Item>? items,
    int? dateAdded,
  }) {
    return Question(
      id: id ?? this.id,
      checkList: checkList ?? this.checkList,
      items: items ?? this.items,
      dateAdded: dateAdded ?? this.dateAdded,
    );
  }
}

File: Hospital app/app/src/main/java/com/elsharif/hospitalapp/dataofchecklist/Question.kt
package com.elsharif.hospitalapp.dataofchecklist

import androidx.compose.foundation.MutatePriority
import androidx.room.Entity
import androidx.room.PrimaryKey
import androidx.room.TypeConverter
import androidx.room.TypeConverters
import com.google.gson.Gson
import com.google.gson.reflect.TypeToken
import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale

@Entity
@TypeConverters(ListConverter::class)
data class Question(
    var checkList: String,
    var items: List<Item>,
    var name:String,
    var hospital:String,
    var sub:String,
    var priority: Int,
    var dateAdded: Long,
    var score:Double,
    @PrimaryKey(autoGenerate = true)
    val id: Int = 0
){
    fun formattedDateTime(): String {
        val sdf = SimpleDateFormat("dd/MM/yyyy hh:mm a", Locale.getDefault())
        val date = Date(dateAdded)
        return sdf.format(date)
    }
}

data class Item(
    val title: String,
    var subItems: List<Answer>
)
data class Answer(
    val question: String,
    val answer: String
)


class ListConverter {
    @TypeConverter
    fun fromString(value: String): List<Item> {
        val listType = object : TypeToken<List<Item>>() {}.type
        return Gson().fromJson(value, listType)
    }

    @TypeConverter
    fun toString(list: List<Item>): String {
        val gson = Gson()
        return gson.toJson(list)
    }
}